### PR TITLE
fix: consolidate use of api

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

You should be proud of me; I added some tests 😁

### CODE REVIEW

File: api-server/src/server/boot/challenge.js [hunk 1]
```diff
@@ -82,6 +78,295 @@ export default async function bootChallenge(app, done) {
...
+  // TODO: Handle resubmission of projects (no new timestamp)
+  const updatedProperties = {};
+}
+
+/**
+ * From Naomi:
+ * TODO: Shaun, we'll need to update this to not use the challenge types from the client.
+ * We want to decouple the API and client, and this recouples it. Maybe a question for Oliver, but
+ * can we move the challengeTypes stuff into the root level utils?? Otherwise, we'll need to hard code here
+ * IMHO.
```
Comment (ShaunSHamilton) [inline api-server/src/server/boot/challenge.js:268]: @ojeytonwilliams Shall I hard-code this now? Or, move the challengeTypes to `/utils`?

Comment (ojeytonwilliams) [reply api-server/src/server/boot/challenge.js]: Moving it to `/utils` makes the most sense to me. We're already compiling any ts files in there, so it should be fine.

**Comment by ShaunSHamilton** [comment]
Close, but we think Loopback is sabotaging us:

```console
MongoError: Updating the path 'completedChallenges' would create a conflict at 'completedChallenges'
    at Connection.<anonymous> (/home/shauh/freeCodeCamp/node_modules/mongodb/lib/core/connection/pool.js:453:61)
    at Connection.emit (node:events:390:28)
    at Connection.emit (node:domain:475:12)
    at processMessage (/home/shauh/freeCodeCamp/node_modules/mongodb/lib/core/connection/connection.js:456:10)
    at TLSSocket.<anonymous> (/home/shauh/freeCodeCamp/node_modules/mongodb/lib/core/connection/connection.js:625:15)
    at TLSSocket.emit (node:events:390:28)
    at TLSSocket.emit (node:domain:475:12)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:199:23) {
  ok: 0,
  code: 40,
  codeName: 'ConflictingUpdateOperators',
  '$clusterTime': {
    clusterTime: Timestamp { _bsontype: 'Timestamp', low_: 6, high_: 1652985813 },
    signature: {
      hash: Binary {
        _bsontype: 'Binary',
        sub_type: 0,
        position: 20,
        buffer: Buffer(20) [Uint8Array] [
           53, 228,  12, 104,  86, 191,
          219, 173, 203, 240, 200, 101,
          131, 251, 212,  26,  46, 100,
            9, 223
        ]
      },
      keyId: Long { _bsontype: 'Long', low_: 1000, high_: 1640880794 }
    }
  },
  operationTime: Timestamp { _bsontype: 'Timestamp', low_: 6, high_: 1652985813 }
}
```

**Comment by ShaunSHamilton** [comment]
@ojeytonwilliams I have opened a new PR with the new endpoints in the API

Once that is in, I will change this PR into the one with the client changes, and add the final PR to deprecate.

**Comment by raisedadead** [comment]
We are officially on Node.js 18 now. We have changed the merge requirements accordingly.

You will need to rebase this PR against the main branch OR Click the "Update branch" button to get the new checks going.

Thanks for your contribution.

**Comment by raisedadead** [comment]
Closing stale, you should bookmark this and re-open a fresh PR when this becomes a priority

**Pull request closed without merging**